py/objstr.c

```diff
@@ -1285,6 +1297,30 @@ mp_obj_t mp_obj_str_format(mp_uint_t n_args, const mp_obj_t *args, mp_map_t *kwa
         }
     }
 
+    if (str <= top) {
+        int count = str - start - 1;
+        if (!level) {
+            count += 1;
+        }
+        vstr_cut_out_bytes(str_input, position, count);
```

Comment:
What you do in these 2 lines is moving memory in one direction, just move it on another after that. Instead, that should be done once.